### ISSUE TITLE
Terminal adding a space between characters

### ISSUE THREAD

**Issue by dmmartins** [cdr/code-server#4204]
## OS/Web Information

- Web Browser: Chrome Version 93.0.4577.82 (Official Build) (x86_64)
- Local OS: macOS 10.15.7
- Remote OS: Linux 4.19.0-16-cloud-amd64  Debian 4.19.181-1
- Remote Architecture: amd64
- `code-server --version`: 3.12.0 b37ff28a0a582aee84a8f961755d0cb40a4081db

## Steps to Reproduce

Couldn't find a way to reproduce yet, but it's happening at least once a day. After a terminal is open for a while, the terminal looks like the screenshot attached

## Screenshot

<img width="1094" alt="Screen Shot 2021-09-18 at 13 07 48" src="https://user-images.githubusercontent.com/176215/133895398-cc6928b7-b68f-47c6-8f40-3a203ecc061a.png">

## Notes

This issue can be reproduced in VS Code: No

Also this didn't happen on previous versions. Started happening after I upgraded to 3.12.0

**Comment by jsjoeio** [comment]
Oh wow...I wonder if that's font rendering issue or something?

Are you using a custom font?

**Comment by dmmartins** [comment]
No, default chrome fonts. It didn't happen again since I open the ticket, I'll check if there's actually a space char there or is just rendering when it happens again.

**Comment by jsjoeio** [comment]
> No, default chrome fonts.

Sorry, I meant have you specified a font or changed anything related to fonts in `settings.json` inside code-server?

> It didn't happen again since I open the ticket

Well, that's good! Okay, I'll let you close this then after checking again.

**Comment by dmmartins** [comment]
> I meant have you specified a font or changed anything related to fonts in settings.json inside code-server?

default font as well.

I found a way to reproduce it:

1. open the editor and open a terminal
2. reload the page

resizing the page will fix it

**Comment by jsjoeio** [comment]
> I found a way to reproduce it:

Hmm...I'm on macOS + Brave and followed those instructions. I can't reproduce though.

https://user-images.githubusercontent.com/3806031/134388783-5dd92a78-9a8f-43c6-ad36-1a823cde7a87.mov

Tried Chrome as well. No luck. I wonder if it's a browser setting?

https://user-images.githubusercontent.com/3806031/134389029-b67b9b3b-654c-4b67-a7c6-87741df9ea25.mov

**Comment by rednaks** [comment]
I think this bug was reported several times and it really needs to be investigated. The workaround is the set the font in the config and don't keep it empty ..

https://github.com/cdr/code-server/issues/3848#issuecomment-886882393

**Comment by jsjoeio** [comment]
@rednaks unfortunately we can't investigate if we can't reproduce this issue.

In the past, it's usually custom font or browser-related. If you can provide specific instructions for reproducing in code-server, I'm happy to take a look!

Otherwise, it could be an issue with the browser, custom font, or [xterm.js](https://xtermjs.org/)

**Comment by rednaks** [comment]
@jsjoeio well that's interesting, I just opened xterm,js on my browser and got the same issue, I'm using firefox. on chromium it looks good. How can I investigate this further ? 

![image](https://user-images.githubusercontent.com/1383161/134819316-5eddd2f2-fd12-4d6e-9bc7-2e174946b2d7.png)

**Comment by jsjoeio** [comment]
Wow! Thanks for sharing @rednaks! Indeed, this looks to be an issue with xterm then.

> How can I investigate this further 

Great question! If I were you, I'd open an issue under [their repo](https://github.com/xtermjs/xterm.js/issues/new/choose) and see what they say. I took a brief look and [didn't see anything](https://github.com/xtermjs/xterm.js/issues?q=is%3Aissue+is%3Aopen+space+label%3Abrowser%2Ffirefox+) already open (but I'd double-check).

**Comment by jsjoeio** [comment]
And if you don't mind @rednaks, can you link it here? Then I can mark this upstream and close

**Comment by bpmct** [comment]
It’s possible this has something to do with your host operating system’s locale and encoding, which makes it run weirdly in VS Code?

I have also seen terminal sessions in code-server have this issue, but haven’t been able to reproduce consistently.
